File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mpp;
        for(int i=0;i<nums.size();i++){
            int firstDigit=nums[i];
            int remaing=target-firstDigit;
            if(mpp.find(remaing)!=mpp.end()){
                return {mpp[remaing],i};
            }
            mpp[firstDigit]=i;
        }
        return {-1,-1};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numap={};
    for(let i=0;i<nums.length;++i){
        const complement = target- nums[i];
        if(numap.hasOwnProperty(complement)){
            return [numap[complement],i];
        }
        numap[nums[i]]=i;
    }
};